addons/NoCampaignMusic/NoCampaignMusic.lua

```diff
@@ -0,0 +1,168 @@
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'No Campaign Music'
+_addon.author = 'Dean James (Xurion of Bismarck)'
+_addon.version = '2.0.0'
+_addon.commands = {'nocampaignmusic', 'ncm'}
+
+packets = require('packets')
+config = require('config')
+
+defaults = {
+    active = true,
+    notifications = false,
+}
+
+settings = config.load(defaults)
+
+campaign_id = 247
+solo_id = 101
+party_id = 215
+solo_dungeon_id = 115
+party_dungeon_id = 216
+
+campaign_active = false
+
+zone_music_map = {}
+zone_music_map[80] = { 254, 254, solo_id, party_id } --Southern San d'Oria [S]
+zone_music_map[81] = { 251, 251, solo_id, party_id } --East Ronfaure [S]
+zone_music_map[82] = { 0, 0, solo_id, party_id } --Jugner Forest [S]
+zone_music_map[83] = { 0, 0, solo_id, party_id } --Vunkerl Inlet [S]
+zone_music_map[84] = { 252, 252, solo_id, party_id } --Batallia Downs [S]
+zone_music_map[85] = { 44, 44, solo_dungeon_id, party_dungeon_id } --La Vaule [S]
+zone_music_map[87] = { 180, 180, solo_id, party_id } --Bastok Markets [S]
+zone_music_map[88] = { 253, 253, solo_id, party_id } --North Gustaberg [S]
+zone_music_map[89] = { 0, 0, solo_id, party_id } --Grauberg [S]
+zone_music_map[90] = { 0, 0, solo_id, party_id } --Pashhow Marshlands [S]
+zone_music_map[91] = { 252, 252, solo_id, party_id } --Rolanberry Fields [S]
+zone_music_map[92] = { 44, 44, solo_dungeon_id, party_dungeon_id } --Beadeaux [S]
+zone_music_map[94] = { 182, 182, solo_id, party_id } --Windurst Waters [S]
+zone_music_map[95] = { 141, 141, solo_id, party_id } --West Sarutabaruta [S]
+zone_music_map[96] = { 0, 0, solo_id, party_id } --Fort Karugo-Narugo [S]
+zone_music_map[97] = { 0, 0, solo_id, party_id } --Meriphataud Mountains [S]
+zone_music_map[98] = { 252, 252, solo_id, party_id } --Sauromugue Champaign [S]
+zone_music_map[99] = { 44, 44, solo_dungeon_id, party_dungeon_id } --Castle Oztroja [S]
+zone_music_map[136] = { 0, 0, solo_id, party_id } --Beaucedine Glacier [S]
+zone_music_map[137] = { 42, 42, solo_id, party_id } --Xarcabard [S]
+zone_music_map[138] = { 43, 43, solo_dungeon_id, party_dungeon_id } --Castle Zvahl Baileys [S]
+zone_music_map[155] = { 43, 43, solo_dungeon_id, party_dungeon_id } --Castle Zvahl Keep [S]
+zone_music_map[164] = { 0, 0, solo_dungeon_id, party_dungeon_id } --Garlaige Citadel [S]
+zone_music_map[171] = { 0, 0, solo_dungeon_id, party_dungeon_id } --Crawlers' Nest [S]
+zone_music_map[175] = { 0, 0, solo_dungeon_id, party_dungeon_id } --The Eldieme Necropolis [S]
+
+windower.register_event('incoming chunk', function(id, data)
+    if id == 0x00A then --Zone update (zoned in)
+        local parsed = packets.parse('incoming', data)
+        if parsed['Day Music'] == campaign_id and zone_music_map[parsed.Zone] then
+            campaign_active = true
+            if not settings.active then return end
+
+            parsed['Day Music'] = zone_music_map[parsed.Zone][1]
```

Comment:
I would cache the lookup:
```lua
local zone_music = zone_music_map[parsed.Zone]
```

Would make it a bit easier to read, and even improve performance ever so slightly :) And the same in the lower packet. In fact, could remove a bit of code duplication if you did this:
```lua
if id ~= 0x00A and id ~= 0x05F then
    return
end

local parsed = packets.parse('incoming', data)
local zone_music = zone_music_map[parsed['Zone'] or windower.ffxi.get_info().zone]

if id == 0x00A then
    -- Zone in packet logic
else
    -- Music change packet logic
end
```

But some people would probably find it a bit over-engineered, so I'll leave it up to you.

Reply:
I like it! Thanks for taking the time to write it 👌